File: Compiler/src/compiler/values/RelationshipValue.java
package compiler.values;

import compiler.Value;
import dictionaries.RelationshipsDictionary;
import util.CompilationResult;
import util.DataType;
import util.JenaUtil;

import static util.JenaUtil.POAS_PREF;

public class RelationshipValue extends Value {
    public RelationshipValue(String value) {
        super(value);
        if(!RelationshipsDictionary.exist(value)) throw new IllegalArgumentException("Указанного отношения не существует");
    }

    @Override
    public DataType resultDataType() {
        return DataType.RELATIONSHIP;
    }

    @Override
    public CompilationResult compile() {
        return new CompilationResult(JenaUtil.genRuleLink(POAS_PREF, value), "", "");
    }
}

File: Compiler/src/compiler/values/RelationshipValue.kt
package compiler.values

import compiler.Value
import dictionaries.RelationshipsDictionary
import util.CompilationResult
import util.DataType
import util.JenaUtil
import util.JenaUtil.POAS_PREF

class RelationshipValue(value: String) : Value(value) {

    init {
        require(RelationshipsDictionary.exist(value)) { "Указанного отношения не существует" }
    }

    override fun resultDataType(): DataType = DataType.Relationship

    override fun compile(): CompilationResult =
        CompilationResult(JenaUtil.genLink(POAS_PREF, value), emptyList(), "")
}